src/main/java/core/basesyntax/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(Storage storage, FruitTransaction transaction) {
+        int currentQuantity = storage.getQuantity(transaction.getFruit());
+        int newQuantity = currentQuantity - transaction.getQuantity();
+
+        if (newQuantity < 0) {

```

Comment:
According to the checklist, in the handling of the Purchase operation, it's required to check the result balance before saving it in the Storage and ensure it is positive. You've implemented this check correctly.